src/execution_plan/record.c

```diff
@@ -111,6 +113,91 @@ size_t Record_ToString(const Record r, char **buf, size_t *buf_cap) {
     return SIValue_StringConcat(values, rLen, *buf, *buf_cap);
 }
 
+unsigned long long Record_Hash64(const Record r) {
+    uint rec_len = Record_length(r);
+    void *data;
+    size_t len;
+    static long long _null = 0;
+    GraphEntity *gent;
+    struct {
+        GraphEntityType type;
+        EntityID id;
+    } entity;
+    SIValue si;
+    
+	XXH_errorcode res;
+    XXH64_state_t* const state = XXH64_createState();
+    if (!state) abort();
+
+    res = XXH64_reset(state, 0);
+    if (res == XXH_ERROR) abort();
+    
+    bool finish = false;
+    for(int i = 0; i < rec_len; ++i) {
+        Entry e = r[i];
+        switch(e.type) {
+        case REC_TYPE_NODE:
+            gent = (GraphEntity*) Record_GetNode(r, i);
+            entity.type = GETYPE_NODE;
+            entity.id = ENTITY_GET_ID(gent);
+            data = &entity;
+            len = sizeof(entity);
+            break;
+            
+        case REC_TYPE_EDGE:
+            gent = (GraphEntity*)Record_GetEdge(r, i);
+            entity.type = GETYPE_EDGE;
+            entity.id = ENTITY_GET_ID(gent);
+            data = &entity;
+            len = sizeof(entity);
+            break;
+            
+        case REC_TYPE_SCALAR:
+            si = Record_GetScalar(r, i);
+            switch (si.type) {
+            case T_NULL:
+                data = &_null;
+                len = sizeof(_null);
+                break;
+                
+            case T_STRING:
+            case T_CONSTSTRING:
+                data = si.stringval;
+                len = si.stringval ? strlen(si.stringval) : 0;
+                break;
+                
+            case T_INT64:
+            case T_BOOL:
+            case T_PTR:
+                data = &si;
+                len = sizeof(si.longval);
+                break;
+                
+            default:
+                abort();
+            }
+            break;
+
+        case REC_TYPE_UNKNOWN:
+            abort();
+            break;
+
+        default:
+            abort();
+        }
+
+        if (finish)
+            break;
+
+        res = XXH64_update(state, data, len);
+        if (res == XXH_ERROR) abort();
+    }
+
+    unsigned long long const hash = XXH64_digest(state);
+    XXH64_freeState(state);
```

Comment:
XXH64_createState() and XXH64_freeState() actually use malloc to allocate a XXH64_state_t. We can use an automatic one.
XXH64_reset() does some work to initialize XXH64_state_t that we can replace with a single memcpy, but it may not be a large gain.